### PR TITLE
Rework casting IntPtr by using ParamToInt

### PR DESCRIPTION
**Pull request by DmitryGorokhov**
Fixes #

## Proposed changes

- Rework cast using `(int)` and `ToInt32` constructions to `PARAM.ToInt` method

## Customer Impact

- No

## Regression? 

- No

## Risk

- Minimal

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5203)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs [hunk 1]
```diff
@@ -1823,7 +1823,7 @@ private void SetRange(DateTime minDate, DateTime maxDate)
                 sa[0] = DateTimePicker.DateTimeToSysTime(minDate);
                 sa[1] = DateTimePicker.DateTimeToSysTime(maxDate);
                 GDTR flags = GDTR.MIN | GDTR.MAX;
-                if ((int)User32.SendMessageW(this, (User32.WM)MCM.SETRANGE, (IntPtr)flags, ref sa[0]) == 0)
+                if (PARAM.ToInt(User32.SendMessageW(this, (User32.WM)MCM.SETRANGE, (IntPtr)flags, ref sa[0])) == 0)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.cs:1826]:
```suggestion
                if (User32.SendMessageW(this, (User32.WM)MCM.SETRANGE, (IntPtr)flags, ref sa[0]) == IntPtr.Zero)
```

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs [hunk 2]
```diff
@@ -244,7 +244,7 @@ public Rectangle Bounds
                 unsafe
                 { *((IntPtr*)&rc.left) = Handle; }
                 // wparam: 1=include only text, 0=include entire line
-                if ((int)User32.SendMessageW(tv, (User32.WM)TVM.GETITEMRECT, (IntPtr)1, ref rc) == 0)
+                if (PARAM.ToInt(User32.SendMessageW(tv, (User32.WM)TVM.GETITEMRECT, (IntPtr)1, ref rc)) == 0)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:247]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs [hunk 3]
```diff
@@ -273,7 +273,7 @@ internal Rectangle RowBounds
                     return Rectangle.Empty;
                 }
 
-                if ((int)User32.SendMessageW(tv, (User32.WM)TVM.GETITEMRECT, IntPtr.Zero, ref rc) == 0)
+                if (PARAM.ToInt(User32.SendMessageW(tv, (User32.WM)TVM.GETITEMRECT, IntPtr.Zero, ref rc)) == 0)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:276]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs [hunk 4]
```diff
@@ -677,7 +677,7 @@ public bool IsVisible
                 unsafe
                 { *((IntPtr*)&rc.left) = Handle; }
 
-                bool visible = ((int)User32.SendMessageW(tv, (User32.WM)TVM.GETITEMRECT, (IntPtr)1, ref rc) != 0);
+                bool visible = (PARAM.ToInt(User32.SendMessageW(tv, (User32.WM)TVM.GETITEMRECT, (IntPtr)1, ref rc)) != 0);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:680]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeView.cs [hunk 5]
```diff
@@ -1979,14 +1979,14 @@ protected override void OnHandleCreated(EventArgs e)
...
-                int style = unchecked((int)User32.GetWindowLong(this, User32.GWL.STYLE));
+                int style = unchecked(PARAM.ToInt(User32.GetWindowLong(this, User32.GWL.STYLE)));
                 style |= (int)TVS.CHECKBOXES;
                 User32.SetWindowLong(this, User32.GWL.STYLE, (IntPtr)style);
             }
 
             if (ShowNodeToolTips && !DesignMode)
             {
-                int style = unchecked((int)User32.GetWindowLong(this, User32.GWL.STYLE));
+                int style = unchecked(PARAM.ToInt(User32.GetWindowLong(this, User32.GWL.STYLE)));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeView.cs:1989]:
```suggestion
                int style = PARAM.ToInt(User32.GetWindowLong(this, User32.GWL.STYLE));
```

**Review by RussKie — CHANGES_REQUESTED** [review]
👍

**Review by vladimir-krestov — DISMISSED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs (hunk 4 above)
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs:680]: Compare with IntPtr.Zero

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeView.cs [hunk 6]
```diff
@@ -1979,14 +1979,14 @@ protected override void OnHandleCreated(EventArgs e)
             // This seems to make the Treeview happy.
             if (CheckBoxes)
             {
-                int style = unchecked((int)User32.GetWindowLong(this, User32.GWL.STYLE));
+                int style = unchecked(PARAM.ToInt(User32.GetWindowLong(this, User32.GWL.STYLE)));
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeView.cs:1982]: remove `unchecked`

**Review by RussKie — COMMENTED** [review]
👍 modulo the outstanding comments

**Review by RussKie — APPROVED** [review]

**Pull request merged**